pandas/io/stata.py

```diff
@@ -1521,42 +1516,64 @@ def _read_value_labels(self) -> None:
             self._path_or_buf.seek(self._data_location + offset)
 
         self._value_labels_read = True
-        self._value_label_dict = {}
+        self._value_label_dict: dict[str, dict[int, str]] = {}
 
         while True:
             if self._format_version >= 117:
                 if self._path_or_buf.read(5) == b"</val":  # <lbl>
                     break  # end of value label table
 
-            slength = self._path_or_buf.read(4)
-            if not slength:
-                break  # end of value label table (format < 117)
-            if self._format_version <= 117:
-                labname = self._decode(self._path_or_buf.read(33))
+            if self._format_version >= 108:
+                slength = self._path_or_buf.read(4)
+                if not slength:
```

Comment:
Perhaps this should be `if self._format_version < 117 and not slength:` so that it only checks on 108 <= x < 117

Reply:
I hadn't looked at the slength bit previously, but it could probably be a bit clearer. The `if self._format_version >= 117:` lines should mean that for a well-formed file the loop breaks before `if not slength:` becomes true, however for a case where `</val` is missing or in the wrong place the while loop would not exit if I added `< 117` to the `is not slength` condition. Basically, as I understand it, this is allowing the loop to end if an end-of-file is reached. In versions < 117 this is how the end of the value label section is indicated, but in 117 and greater this should be indicted with the string </value_labels>.